### PR TITLE
OK-2568-optimize error tips when updating firmware

### PR DESCRIPTION
**Pull request by liyanhrxy**
## What does this implement/fix? Explain your changes.
optimize error tips when updating firmware

## Does this close any currently open issues?  
[OK-2568] [OK-2649]

## Pull request type
 
_Put an `x` in the boxes that apply_
- [] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [x] Other (please describe): 

## Where has this been tested?
local

## Any other comments?
none

[OK-2568]: https://onekeyhq.atlassian.net/browse/OK-2568
[OK-2649]: https://onekeyhq.atlassian.net/browse/OK-2649

### CODE REVIEW

File: electrum_gui/android/hardware.py [hunk 1]
```diff
@@ -399,9 +404,18 @@ def firmware_update(
...
+        except Exception as e:
+            if "PIN" in str(e):
+                raise exceptions.HardwareInvalidPIN() from e
+            else:
+                raise exceptions.HardwareUpdateFailed() from e
+        else:
+            if not is_bootloader:
+                # reboot returned normal with initial state
+                raise exceptions.HardwareUpdateFailed(_("Upgrade failed due to invalid state"))
+        # raw is always False currently currently
```
Comment (taimanhui) [inline electrum_gui/android/hardware.py:418]: 多了个`currently`

File: electrum_gui/android/hardware.py [hunk 2]
```diff
@@ -412,28 +426,7 @@ def firmware_update(
             data = data[256:]
 
         if skip_check:
```
Comment (taimanhui) [inline electrum_gui/android/hardware.py:428]: 这个没用了，不能去掉嘛？

**Review by taimanhui — COMMENTED** [review]
github的描述连jira可以直接输`[OK-2568]`这样子。 [OK-2568]

File: electrum_gui/android/hardware.py (hunk 2 above)
Comment (liyanhrxy) [reply electrum_gui/android/hardware.py:428]: 一直都是走的这块，下面的被我删了

File: electrum_gui/android/hardware.py (hunk 1 above)
Comment (liyanhrxy) [reply electrum_gui/android/hardware.py:418]: > github的描述连jira可以直接输`[OK-2568]`这样子。 [OK-2568]

不是说要写在标题上吗？

Comment (taimanhui) [reply electrum_gui/android/hardware.py]: 不是，我说的是上面描述那里，标题还是要写...

**Comment by taimanhui** [comment]
描述上面那条review的评论不认...好矬...

[OK-2568]

**Review by taimanhui — APPROVED** [review]

File: electrum_gui/android/hardware.py (hunk 2 above)
Comment (liyanhrxy) [reply electrum_gui/android/hardware.py:428]: 嗯

Comment (huazhouwang) [inline electrum_gui/android/hardware.py:428]: 这里的 skip_check 其实没意义了吧？

**Review by huazhouwang — APPROVED** [review]

**Pull request merged**